### PR TITLE
6239-fix-remote-ip-logging-and-trusted-proxy-config

### PR DESCRIPTION
**Pull request by ttoomey**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

[GH Issue
](https://github.com/open-path/Green-River/issues/6239)

This PR accomplishes two things:
1) Fixes missing logging information in some requests. All controllers that inherit directly from base controller now include a method to populate the request ips. Also populates xff and remote_addr
2) Adds trusted proxy to correctly extract remote ip from the xff header

## Type of change
[//]: # 'remove options that are not relevant'
- [x] New feature (adds functionality)

## Checklist before requesting review
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [x] My code follows the style guidelines of this project (rubocop)
- [x] I have updated the documentation (or not applicable)
- [x] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

File: config/application.rb
Comment (eanders) [inline config/application.rb]: Should we limit this to staging environments just to be extra careful?

**Review by eanders — APPROVED** [review]

**Comment by ttoomey** [comment]
@eanders could we just deploy this branch to QA to test it out (without merging it)

**Comment by ttoomey** [comment]
@eanders I think I have our logging and proxy situation corrected, just waiting on confirmation from devops that the proxy ip is all that's needed. I took out the middleware I had been using to debug the issue

File: config/application.rb
Comment (eanders) [inline config/application.rb]: This definitely needs to be moved to ENV (or db?)

**Review by eanders — COMMENTED** [review]
Thanks @ttoomey.  This looks great, though I think we should go ahead and get the trusted proxy into the ENV, or at the very least do an `ENV.fetch` where that's the default if the ENV isn't set.

**Comment by ttoomey** [comment]
@eanders adjusted our initialization for proper env support for trusted proxies

**Review by eanders — APPROVED** [review]
Thanks @ttoomey !  Do we need to wait on merging until the ENV and CSV files are deployed?

**Comment by ttoomey** [comment]
> Thanks @ttoomey ! Do we need to wait on merging until the ENV and CSV files are deployed?

I'm going to merge this now, I think it's safe since it won't add trusted proxies unless the env is set. We do need to wait for the env before merging the rack-attack change though

**Comment by eanders** [comment]
Sounds good.  Thanks!

**Pull request merged**
